Fix Windows tests

iangmaia opened the pull request:
As reported in https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/195, Windows unit tests started failing, presumably after all the changes in our Windows AMI.
This PR should fix the Windows unit tests.

iangmaia committed:
Fix windows test

iangmaia committed:
Fix SkipWindows10SDKInstallation flag precedence

iangmaia committed:
Set AMI_VERSION to 0.1 in tests 3 and 4 to avoid early exit and test SDK file logic

iangmaia committed:
Update CHANGELOG

AliSoftware commented on bin/prepare_windows_host_for_app_distribution.ps1:
```diff
@@ -27,21 +27,29 @@ param (
+    Exit 0
+  }
+}
+
```
Those changes don't seem right nor necessary.

The whole point of testing the `AMI_VERSION` as early as possible in this script is that we can exit early and make this script mostly a no-op if it was called by a repo and pipeline that hasn't been updated yet (to stop calling this script altogether) after we deployed the custom AMI for the Windows queue that contains all the pre-installed tooling. In that case, we don't even care if the parameter combinations or the `SkipWindows10SDKInstallation` parameters were set or not.

In the medium term this script will completely disappear and we don't expect any repo to call it anymore now that the custom AMI has been deployed anyway. This early exit on `AMI_VERSION>=2` was thus to help with the transition, so that while we're working on this and have already deployed the custom AMI on our `windows` queue in CI, any repo that is still calling this `prepare_windows_host_for_app_distribution.ps1` (because we haven't had the time to update the `.buildkite/commands/*.sh` scripts of said repo just yet) would result in those being an almost no-op (modulo the installation of the signing certificate, which for good security reasons is not baked into the custom AMI). In the long run we plan to update all those `.buildkite/commands/*.sh` of all the repos that were calling this to just call `setup_windows_code_signing.ps1` directly instead, and once it's done, we'll delete the `prepare_windows_host_for_app_distribution.ps1` from the `release-toolkit` entirely.

---

In fact, we could even consider deleting that `prepare_windows_host_for_app_distribution.ps1` script from the `release-toolkit` repo right away right now, given that any repo that is still using it (because we haven't had the time to update their `.buildkite/commands/*.sh` just yet) is still pointing to `release-toolkit` version `5.7.0` which still has it (and the whole point of this conditional was to give us the necessary time to update the repos to latest release-toolkit version + `setup_windows_code_signing.ps1` calls all in one go)

iangmaia replied:
> In the medium term this script will completely disappear 

> we could even consider deleting that `prepare_windows_host_for_app_distribution.ps1` script [...] right away right now, given that any repo that is still using it [...] is still pointing to release-toolkit version 5.7.0

Ohhh yes indeed. I think I confused myself a bit with what's in the AMI (and if we'd get something keeping the script) -- I agree with removing it straight away to avoid confusions / failing tests given apps will stop using it. Updated on ce15769753ac54dde3b23f63d7c6ad4a9bd209b3.

AliSoftware reviewed (commented):
Overall my suggestion for the fix is to just completely delete the `prepare_windows_host_for_app_distribution.ps1` and its test from the repo.

Once we'll find the time to update all the repos that were still calling `prepare_windows_host_for_app_distribution.ps1` in their `.buildkite/commands/*.sh` scripts and relied on the assumption that they might still be running on the base AMI and not our custom one, we'll have those repos just call `setup_windows_code_signing.ps1` directly instead and bump their `release-toolkit` version to `6.0.0`, at which point no repo using `release-tookit >5.7.0` will call `prepare_windows_host_for_app_distribution.ps1` and it won't be needed anymore (which was the whole point of creating our custom AMI in the first place after all.

iangmaia committed:
Remove `prepare_windows_host_for_app_distribution.ps1` and its tests

h/t @AliSoftware https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/196#discussion_r2543801520 -- this is now done in the AMI itself. Since this plugin is versioned and apps use a pinned version, we can safely remove the scripts.

AliSoftware commented on CHANGELOG.md:
```diff
@@ -46,7 +46,7 @@ _None._
 ### Internal Changes
 
-_None._
+- Fixed Windows unit tests after Windows AMI changes. [#196]
```
```suggestion
_None._
```

AliSoftware commented on CHANGELOG.md:
```diff
@@ -34,7 +34,7 @@ _None._
 ### Breaking Changes
 
-_None._
+- We've removed the `prepare_windows_host_for_app_distribution.ps1` script as this functionality is now implemented in the Windows AMI itself. [#196]
```
I don't think we ever used the "We've …" form/wording in any of our other changelog entries in the past?

```suggestion
- Removed the `prepare_windows_host_for_app_distribution.ps1` script as what it was installing is now already pre-provisioned in our latest custom Windows AMI already. [#196]
```

iangmaia committed:
Update CHANGELOG.md

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

AliSoftware commented on CHANGELOG.md:
```diff
@@ -34,7 +34,7 @@ _None._
 ### Breaking Changes
 
-_None._
+- Removed the `prepare_windows_host_for_app_distribution.ps1` script as what it was installing is now already pre-provisioned in our latest custom Windows AMI already. [#196]
```
Whoops repetition of `already`

```suggestion
- Removed the `prepare_windows_host_for_app_distribution.ps1` script, given what it was installing is now already pre-provisioned in our custom Windows AMI. [#196]
```

AliSoftware committed:
Apply suggestion from @AliSoftware
